common-arcana.lic

```diff
@@ -234,6 +234,34 @@ module DRCA
       .map { |_name, properties| properties['abbrev'] }
       .each { |abbrev| fput("release #{abbrev}") }
   end
+  
+  def parse_regalia  # generates an array of currently-worn regalia armor nouns
+    return unless DRStats.trader?
+    DRC.bput('inv armor', 'Type INVENTORY HELP for more options')
+    snapshot = reget(40)
+    if snapshot.grep(/All of your armor|You aren't wearing anything like that/).any? && snapshot.grep(/Type INVENTORY HELP/).any?
+      result = snapshot
+        .map(&:strip)
+        .reverse
+        .take_while { |item| !['All of your armor:', "You aren't wearing anything like that."].include?(item) }
+        .drop_while { |item| item != '[Type INVENTORY HELP for more options]' }
+        .drop(1)
+        .select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal')}
+    else
+      parse_regalia
+    end
+    result.each_with_index { |item, index| result[index] = DRC.get_noun(item)}
```

Comment:
💡 Is this the same as `result.map { |item| DRC.get_noun(item) }`?

Reply:
Thanks, I knew there had to be a better way to do that.  I can move that into the if to become
```      
result = snapshot
        .map(&:strip)
        .reverse
        .take_while { |item| !['All of your armor:', "You aren't wearing anything like that."].include?(item) }
        .drop_while { |item| item != '[Type INVENTORY HELP for more options]' }
        .drop(1)
        .select { |item| item.include?('rough-cut crystal') || item.include?('faceted crystal') || item.include?('resplendent crystal')}
        .map { |item| DRC.get_noun(item)}
```